ruler-gradle-plugin/src/main/kotlin/com/spotify/ruler/plugin/report/JsonReporter.kt

```diff
@@ -38,6 +48,7 @@ class JsonReporter {
      * @param appInfo General info about the analyzed app.
      * @param components Map of app component names to their respective files
      * @param ownershipInfo Optional info about the owners of components.
+     * @param shouldExcludeFileListing Flag to determine if file listing should be included in the report
```

Comment:
main change if this PR. I will try to explain the main idea:
- New `Insights` field on `AppReport`: pre-calculated insights about the app (same info shown on "Insights" tab"). It contains the same calculated app sizes and sizes by type of file/component that were previously calculated on `Breakdown.kt` file
- New `ownershipOverview: Map<String, OwnershipOverview>` field on `AppReport`: it holds data about the size of the app that belongs to each owner and the size of their files that doesn't belong to any of their owned component
- New `Owner` class on `FileContainer` interface: it contains the name of the owner of a `FileContainer` and the its corresponding size (it can be equal to the whole size of the component or less if that container contains files owned by other teams)
- Filter out every file from components and dynamic features if `shouldExcludeFileListing` flag is true